src/ast/grammar.peg

```diff
@@ -85,10 +85,9 @@ sym <- <[!-'*+\x2D./:<-@\\^|~]+> # Reserved: , _ () [] {} ` ;
 float <- <digits '.' digits ([eE][-+]? digits)?>
 int <- <digits>
 digits <- [0-9][_0-9]*
-hex <- <'0x' [_a-fA-F]*>
+hex <- <'0x' [_0-9a-fA-F]*>
 binary <- <'0b' [_01]*>
-string <- '`' ('(' <((!'(' !')' .) / substring)*> ')' / <%word>)
-substring <- '(' ('\\)' / (!'(' !')' .) / substring)* ')'
+string <- '`(' <('\\)' / string / (!'(' !')' .))*> ')'
```

Comment:
My peg-foo isn't good enough yet to know the difference. @sylvanc?

Reply:
The previous implementation didn't handle nested strings or the escaped `\)`

Reply:
I'd like to hear @sylvanc's thoughts on this in case I was missing something

Reply:
Just checking my understanding
```
escaped_closing_paren <- '\\)'
not_paren <- (!'(' !')' .)
 '`(' <(escaped_closing_paren / string / not_paren)*> ')'
```
In the [paper](https://bford.info/pub/lang/peg.pdf), for the nested comments, they don't restrict the opening token, just the closing one.  
```
Comment <- ’(*’ (Comment / !’*)’ .)* ’*)’
```
I guess for your implementation this means that
```
`( foo `( )
```
Is not a valid string, and cannot be escaped either?

Reply:
Fixed, you're right. I added the `\(` escape from the original grammar

Reply:
I think the original implementation is correct. It allows balanced parens, ignoring the backtick inside a string. It currently allows escaping unbalanced right parens - perhaps we should allow an unbalanced left paren as well:

```
string <- '`' ('(' <('\\(' / '\\)' / (!'(' !')' .) / substring)*> ')' / <%word>)
substring <- '(' ('\\(' / '\\)' / (!'(' !')' .) / substring)* ')'
```

Reply:
I think we can leave the unbalanced `` `(`` for a future PR